src/lucky_record/model.cr

```diff
@@ -36,6 +39,7 @@ class LuckyRecord::Model
     setup_base_form_class({{table_name}})
     setup_table_name({{table_name}})
     setup_fields_method
+    setup_correct_field_mappings_method
```

Comment:
I think this is a large enough new feature that it would be good to extract a module for this. Something like this maybe:

```crystal
module LuckyRecord::SchemaEnforcer
  macro add_schema_enforcer_methods_for(fields, table_name)
    # generate all the methods that you were previously doing with `macro setup_correct_field_mappings_method`
  end
end
```